src/elf/section_header.rs

```diff
@@ -428,7 +428,13 @@ if_alloc! {
         }
         /// Returns this section header's file offset range
         pub fn file_range(&self) -> Range<usize> {
-            self.sh_offset as usize..(self.sh_offset as usize).saturating_add(self.sh_size as usize)
+            // Sections with type SHT_NOBITS have no data in the file itself,
+            // they only exist in memory.
+            if self.sh_type == SHT_NOBITS {
+                self.sh_offset as usize..self.sh_offset as usize
```

Comment:
is returning a 0 range going to cause issues? ideally we'd return an `Option<Range<usize>>` i think, but that would be breaking change. Might be worth it though?

Reply:
ok i don't think this should be `self.sh_offset..self.sh_offset`; if sh_offset is non-zero you can get e.g., `10..10` or whatever, that is an empty range but will cause a panic if it's larger than buffer, i.e., this panics:
```rust
    let foo = [1, 2, 3];
    let x = &foo[10..10];
```

So i think it's better to return `0..0` ? or Option.

Reply:
Oh, I didn't expect the empty range to panic when it's out of bounds.

I think `0..0`would be confusing as well, if someone looks at the start field of the range to decide where the data in the file starts.

An Option would be the solution, in my opinion.

Reply:
Ok let's make it option, it's the right thing to do :) this will be a breaking change so unless you are in urgent need of a release, i may wait a bit to roll up other potential changes, or make other breaking changes

Reply:
I don't need it urgently, manually checking the section type is an easy workaround.